subsystem-test/tests/src/test/java/org/jboss/as/subsystem/test/transformationutils/TransformationUtilsTestCase.java

```diff
@@ -0,0 +1,483 @@
+        expected.get(pe.getKey(), pe.getValue()).setEmptyObject();
+
+        ModelTestUtils.compare(expected, model);
+    }
+
+
+    @Test
+    public void testHasChildWithAttrReg_EmptySubsystemIsDefined_exclUndefined() throws Exception {
+        // This is one of the cases that https://issues.redhat.com/browse/WFCORE-7036 fixes
+        TransformationUtilsExtension.Builder builder = TransformationUtilsExtension.createBuilder()
+                .addAttribute("test");
+        builder.createChildBuilder(PathElement.pathElement("child", "test"));
+
+        List<ModelNode> operations = new ArrayList<>();
+        operations.add(Util.createAddOperation(SUBSYSTEM_ADDR));
+
+        ModelNode model = runAndReadSubsystemModel(builder, operations, false);
+
+        ModelNode expected = new ModelNode();
+        expected.setEmptyObject();
+
+        ModelTestUtils.compare(expected, model);
+    }
+
+    @Test
+    public void testHasChildWithAttrReg_SubsystemWithAttrIsDefined_exclUndefined() throws Exception {
+        TransformationUtilsExtension.Builder builder = TransformationUtilsExtension.createBuilder()
+                .addAttribute("test");
+        builder.createChildBuilder(PathElement.pathElement("child", "test"));
+
+        List<ModelNode> operations = new ArrayList<>();
+        ModelNode add = Util.createAddOperation(SUBSYSTEM_ADDR);
+        add.get("test").set("one");
+        operations.add(add);
+
+        ModelNode model = runAndReadSubsystemModel(builder, operations, false);
+
+        ModelNode expected = new ModelNode();
+        expected.get("test").set("one");
+
+        ModelTestUtils.compare(expected, model);
+    }
+
+    @Test
+    public void testHasChildWithAttrReg_SubsystemWithChildIsDefined_exclUndefined() throws Exception {
+        TransformationUtilsExtension.Builder builder = TransformationUtilsExtension.createBuilder()
+                .addAttribute("test");
+        PathElement pe = PathElement.pathElement("child", "test");
+        builder.createChildBuilder(pe);
+
+        List<ModelNode> operations = new ArrayList<>();
+        operations.add(Util.createAddOperation(SUBSYSTEM_ADDR));
+        operations.add(Util.createAddOperation(SUBSYSTEM_ADDR.append(pe)));
+
+
+        ModelNode model = runAndReadSubsystemModel(builder, operations, false);
+
+        ModelNode expected = new ModelNode();
+        expected.get(pe.getKey(), pe.getValue()).setEmptyObject();
+
+        ModelTestUtils.compare(expected, model);
+    }
+
+//    @Test
+//    public void testChildRegEmptySubsystemIsDefined() throws Exception {
+//        List<ModelNode> operations = new ArrayList<>();
+//        operations.add(Util.createAddOperation(SUBSYSTEM_ADDR));
+//        TransformationUtilsExtension.Builder builder = TransformationUtilsExtension.createBuilder();
+//        builder.
+//
+//        ModelNode model = runAndReadSubsystemModel(, operations);
+//
+//        Assert.assertTrue(model.isDefined());
+//    }
```

Comment:
@kabir This is an unfinished test, can be removed or was it left pending to get it finished?

Reply:
@yersan I got rid of it